src/transformers/utils/quantization_config.py

```diff
@@ -1038,3 +1038,61 @@ def post_init(self):
         accepted_weights = ["int8"]
         if self.weights not in accepted_weights:
             raise ValueError(f"Only support weights in {accepted_weights} but found {self.weights}")
+
+
+@dataclass
+class GgufConfig(QuantizationConfigMixin):
+    """
+    This is an experimental config class that should be used only when calling `trainer.push_to_hub(gguf_config=gguf_config)`.
+    Do not expect to use it in `AutoModelForxxx.from_pretrained()`.
+
+    Args:
+        quantization_method (`str`):
+            The quantization method to use, supported methods can be found at: https://huggingface.co/spaces/gguf-org/gguf-my-repo
+        space_name (`str`, *optional*, defaults to `"ggml-org/gguf-my-repo"`):
+            Hub ID of the conversion space to use, modify it at your own risk!
+        private (`bool`, *optional*, defaults to `False`):
+            Whether to push the final GGUF file on a private repository.
+        duplicate_space (`bool`, *optional*, defaults to `False`):
+            Whether to duplicate the conversion space or not. This is useful to avoid queue issues if one uses the official
+            Space.
```

Comment:
I really don't think either of these arguments make sense to be included as part of a "quantization config" - they're not configuring quantization, they're configuring a space. It also creates a clash of arguments with the current trainer. If I have private=True in this config and call hub_private_repo=False in the training arguments, which one is should be respected?

I would move these somewhere else (how configurable do we need them to be?)